core/src/main/java/org/apache/calcite/rel/rules/SortJoinCopyRule.java

```diff
@@ -96,9 +96,7 @@ public SortJoinCopyRule(Class<? extends Sort> sortClass,
     if (leftFieldCollation.isEmpty()) {
       newLeftInput = join.getLeft();
     } else {
-      final RelCollation leftCollation =
-          RelCollationTraitDef.INSTANCE.canonize(
-              RelCollations.of(leftFieldCollation));
+      final RelCollation leftCollation = RelCollations.of(leftFieldCollation);
```

Comment:
One question, why did you change it? I observed that you only changed the shift interface, and the shift interface was not called here because this part itself is redundant.

Reply:
[RelCollations.of](https://github.com/apache/calcite/pull/3874/files#diff-d50ce55582ce0f63226670b6234a635b70884a7055fce75df03043b4a77f9f62L85) is already canonizing, only shift needs to be changed to avoid the need of canonization in the callers (this is explained in details in the Jira ticket), so it looks good to me

Reply:
Thanks for your answer, I also noticed this when I looked at the code